src/detect-ftp-command.c

```diff
@@ -59,27 +55,33 @@ static int DetectFtpCommandSetup(DetectEngineCtx *de_ctx, Signature *s, const ch
     return 0;
 }
 
-static InspectionBuffer *GetData(DetectEngineThreadCtx *det_ctx,
-        const DetectEngineTransforms *transforms, Flow *_f, const uint8_t _flow_flags, void *txv,
-        const int list_id)
+static bool DetectFTPCommandGetData(
+        void *txv, const uint8_t _flow_flags, const uint8_t **buffer, uint32_t *buffer_len)
 {
-    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
-    if (buffer->inspect == NULL) {
-        FTPTransaction *tx = (FTPTransaction *)txv;
+    FTPTransaction *tx = (FTPTransaction *)txv;
 
-        if (tx->command_descriptor.command_code == FTP_COMMAND_UNKNOWN)
-            return NULL;
+    if (tx->command_descriptor.command_code == FTP_COMMAND_UNKNOWN)
+        return NULL;
 
-        const char *b = NULL;
-        uint8_t b_len = 0;
-
-        if (SCGetFtpCommandInfo(tx->command_descriptor.command_index, &b, NULL, &b_len)) {
-            InspectionBufferSetupAndApplyTransforms(
-                    det_ctx, list_id, buffer, (const uint8_t *)b, b_len, transforms);
-        }
+    const char *b = NULL;
+    uint8_t b_len = 0;
+    if (SCGetFtpCommandInfo(tx->command_descriptor.command_index, &b, NULL, &b_len)) {
+        *buffer = (const uint8_t *)b;
+        *buffer_len = b_len;
+        return true;
     }
 
-    return buffer;
+    *buffer = NULL;
+    *buffer_len = 0;
+    return false;
+}
+
+static InspectionBuffer *GetDataWrapper(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *_f, const uint8_t _flow_flags, void *txv,
+        const int list_id)
+{
+    return DetectHelperGetData(
```

Comment:
why do we need 2 layers of callbacks?

Reply:
This is a bit on me, see https://github.com/OISF/suricata/pull/13071